src/components/Profile/useUserInfoForm.ts

```diff
@@ -12,14 +13,45 @@ import {
 
 import formatError from 'errors/formatError';
 import { BIO_MAX_CHAR_COUNT } from './UserInfoForm';
+import { fetchQuery, graphql, useRelayEnvironment } from 'react-relay';
+import useDebounce from 'hooks/useDebounce';
+import useAuthPayloadQuery from 'hooks/api/users/useAuthPayloadQuery';
 
 type Props = {
   onSuccess: (username: string) => void;
   existingUsername?: string;
   existingBio?: string;
-  userId: string;
+  userId?: string;
 };
 
+function useIsUsernameAvailableFetcher() {
```

Comment:
the user's cryptographically signed message will not be used immediately to create a new user; instead, it will be forwarded to the onboarding flow where it's submitted once the user decides on a username. since the signature is a one-time key, we want to avoid prompting the user to sign a new nonce upon every user creation attempt. **as a result, we've added real-time username availability checking as the user types into the input, so that when they click "save" it's highly likely that the username is available and their signature is only used once.** this happens to result in an improved UX as well.